### PR TITLE
[v6] remove `MaxItemsOne: true` when appropriate

### PR DESCRIPTION
**Pull request by iwahbe**
I did a case by case analysis of every location where we apply `MaxItemsOne: true`, which restricts the available inputs and outputs that Pulumi can express compared to what ustream can express. 

Fixes https://github.com/pulumi/pulumi-aws/issues/2610

---

## Resources
13 resources
20 overrides to true (this prevents the user from sending multiple values, when they could)
27 overrides to false
### `aws_appmesh_virtual_node`
Justified by [aws_appmesh_virtual_node should support multiple listeners](https://github.com/hashicorp/terraform-provider-aws/issues/26380).
  - `spec.listener`: true
    - `spec.listener.connection_pool.http`: true
  - `spec.listener.connection_pool.http2`: true
  - `spec.listener.connection_pool.tcp`: true
### `aws_appmesh_virtual_router`
Justified by [F aws add multiple listeners and port matching](https://github.com/hashicorp/terraform-provider-aws/pull/27799)
  - `spec.listener`: true
### `aws_appmesh_virtual_gateway`
Justified by [F aws add multiple listeners and port matching](https://github.com/hashicorp/terraform-provider-aws/pull/27799)
  - `spec.listener`: true
### `aws_batch_compute_environment`
Justified by [Fix MaxItems for batch/compute_environments ec2_configuration](https://github.com/hashicorp/terraform-provider-aws/pull/27207)
  - `compute_resources.ec2_configuration`: true
### `aws_ec2_fleet`
Justified by [update aws_ec2_fleet resource to latest functionality](https://github.com/hashicorp/terraform-provider-aws/pull/29181)
  - `launch_template_config`: true
### `aws_eks_cluster`
Not justified: fixed in v5.8.0
  - `certificate_authority`: true
### `aws_iot_topic_rule`
AWS API surface models actions as a list, so this makes sense.
  - `cloudwatch_alarm`: true
  - `cloudwatch_metric`: true
  - `dynamodb`: true
  - `elasticsearch`: true
  - `firehose`: true
  - `kinesis`: true
  - `lambda`: true
  - `republish`: true
  - `s3`: true
  - `sns`: true
  - `sqs`: true
## DataSources
12 datasources
19 overrides to true (this prevents the user from sending multiple values, when they could)
0 overrides to false
### `aws_dynamodb_table`
Not justified, the change seems to be a mistake. (for both)
- `ttl`: true
- `point_in_time_recovery`: true
### `aws_elastic_beanstalk_application`
Not justified: See flattening at https://github.com/hashicorp/terraform-provider-aws/blob/71ac1fa8dd1c0aea46877437921d7443edbe0aa7/internal/service/elasticbeanstalk/application_data_source.go#L73.
- `appversion_lifecycle`: true
### `aws_efs_file_system`
Seems to be a hedge against the future, but not justified. See https://github.com/hashicorp/terraform-provider-aws/commit/362c03ec27e839a571de312060e87657d617038b for details.
- `lifecycle_policy`: true
### `aws_eks_cluster`
Not justified. See https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/eks/cluster_data_source.go#L231 for details.
- `vpc_config`: true
### `aws_lambda_function`
Not justified. See https://github.com/hashicorp/terraform-provider-aws/blob/71ac1fa8dd1c0aea46877437921d7443edbe0aa7/internal/service/lambda/function_data_source.go#L263-L273.
- `dead_letter_config`: true
Not justified. See https://github.com/hashicorp/terraform-provider-aws/blob/71ac1fa8dd1c0aea46877437921d7443edbe0aa7/internal/service/lambda/function_data_source.go#L321-L323.
- `vpc_config`: true
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/71ac1fa8dd1c0aea46877437921d7443edbe0aa7/internal/service/lambda/function_data_source.go#L275-L277.
- `environment`: true
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/71ac1fa8dd1c0aea46877437921d7443edbe0aa7/internal/service/lambda/function_data_source.go#L313-L319.
- `tracing_config`: true
### `aws_mq_broker`
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/mq/broker_data_source.go#L315-L317
- `configuration`: true
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/mq/broker_data_source.go#L336-L338.
- `maintenance_window_start_time`: true
### `aws_elb`
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/elb/load_balancer_data_source.go#L302.
- `access_logs`: true
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/elb/load_balancer_data_source.go#L327
- `health_check`: true
### `aws_lb`
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/elbv2/load_balancer_data_source.go#L319.
- `access_logs`: true
### `aws_lb_target_group`
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/elbv2/target_group_data_source.go#L299.
- `stickiness`: true
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/elbv2/target_group_data_source.go#L231.
- `health_check`: true
### `aws_alb`
Not justified, see https://github.com/hashicorp/terraform-provider-aws/blob/c14a7fe82ab84aaa9db676c9ee4242e20fb33145/internal/service/elbv2/load_balancer_data_source.go#L319.
- `access_logs`: true
### `aws_alb_target_group`
- `health_check`: true
- `stickiness`: true

### CODE REVIEW

**Comment by mnlumi** [comment]
Resolves https://github.com/pulumi/pulumi-aws/issues/2610

File: sdk/nodejs/iot/topicRule.ts [hunk 1]
```diff
@@ -116,20 +116,20 @@ export class TopicRule extends pulumi.CustomResource {
...
     /**
      * The name of the rule.
      */
     public readonly name!: pulumi.Output<string>;
-    public readonly republish!: pulumi.Output<outputs.iot.TopicRuleRepublish | undefined>;
-    public readonly s3!: pulumi.Output<outputs.iot.TopicRuleS3 | undefined>;
-    public readonly sns!: pulumi.Output<outputs.iot.TopicRuleSns | undefined>;
+    public readonly republishes!: pulumi.Output<outputs.iot.TopicRuleRepublish[] | undefined>;
+    public readonly s3s!: pulumi.Output<outputs.iot.TopicRuleS3[] | undefined>;
+    public readonly sns!: pulumi.Output<outputs.iot.TopicRuleSn[] | undefined>;
```
Comment (lukehoban) [inline sdk/nodejs/iot/topicRule.ts:132]: Some of the pluralization/depluralization is really weird here.

File: sdk/nodejs/types/output.ts [hunk 2]
```diff
@@ -33842,7 +33842,7 @@ export namespace iot {
         topic: string;
     }
 
-    export interface TopicRuleKinesis {
+    export interface TopicRuleKinese {
```
Comment (lukehoban) [inline sdk/nodejs/types/output.ts:33845]: Related - very weird/wrong depluralization change here and in a few other places.

Comment (iwahbe) [reply sdk/nodejs/types/output.ts]: Fixed by https://github.com/pulumi/pulumi-aws/pull/2615/commits/743dae9afd4bc455eb0319ca0d26114427fa9e89.

File: sdk/nodejs/iot/topicRule.ts (hunk 1 above)
Comment (iwahbe) [reply sdk/nodejs/iot/topicRule.ts:132]: `s3s` is fixed by https://github.com/pulumi/pulumi-aws/pull/2615/commits/399d2b5866310ea5afdf4e34835cc33c5add045e.

File: sdk/nodejs/types/input.ts [hunk 3]
```diff
@@ -29183,7 +29183,7 @@ export namespace iot {
         roleArn: pulumi.Input<string>;
     }
 
-    export interface TopicRuleSns {
+    export interface TopicRuleSn {
```
Comment (lukehoban) [inline sdk/nodejs/types/input.ts:29186]: Another issue.

File: sdk/nodejs/types/input.ts [hunk 4]
```diff
@@ -29198,7 +29198,7 @@ export namespace iot {
         targetArn: pulumi.Input<string>;
     }
 
-    export interface TopicRuleSqs {
+    export interface TopicRuleSq {
```
Comment (lukehoban) [inline sdk/nodejs/types/input.ts:29201]: Another issue.

File: sdk/nodejs/iot/topicRule.ts [hunk 5]
```diff
@@ -164,25 +164,25 @@ export class TopicRule extends pulumi.CustomResource {
...
             resourceInputs["kafkas"] = state ? state.kafkas : undefined;
-            resourceInputs["kinesis"] = state ? state.kinesis : undefined;
-            resourceInputs["lambda"] = state ? state.lambda : undefined;
+            resourceInputs["kineses"] = state ? state.kineses : undefined;
+            resourceInputs["lambdas"] = state ? state.lambdas : undefined;
             resourceInputs["name"] = state ? state.name : undefined;
-            resourceInputs["republish"] = state ? state.republish : undefined;
-            resourceInputs["s3"] = state ? state.s3 : undefined;
+            resourceInputs["republishes"] = state ? state.republishes : undefined;
+            resourceInputs["s3Buckets"] = state ? state.s3Buckets : undefined;
```
Comment (lukehoban) [inline sdk/nodejs/iot/topicRule.ts:185]: Is it intentional that s3 became s3Buckets?

Comment (iwahbe) [reply sdk/nodejs/iot/topicRule.ts:185]: Yes. I was trying to come up with something that had an understandable plural and the associated object seemed bucket specific (`bucketName` was a field). If we don't have something more specific then that, I can just pin `s3`.

Comment (iwahbe) [reply sdk/nodejs/iot/topicRule.ts]: Fixed (moved back to `s3` with sub-field `s3` in https://github.com/pulumi/pulumi-aws/pull/2615/commits/28e1921c4b75c6e88db8281312d4b817ef3e7825.

File: sdk/nodejs/types/input.ts (hunk 4 above)
Comment (iwahbe) [reply sdk/nodejs/types/input.ts]: Fixed (pinned to `sqs` in https://github.com/pulumi/pulumi-aws/pull/2615/commits/28e1921c4b75c6e88db8281312d4b817ef3e7825.

File: sdk/nodejs/types/input.ts (hunk 3 above)
Comment (iwahbe) [reply sdk/nodejs/types/input.ts]: Fixed (pinned to `sns`) in https://github.com/pulumi/pulumi-aws/pull/2615/commits/28e1921c4b75c6e88db8281312d4b817ef3e7825.

File: sdk/dotnet/Iot/TopicRule.cs [hunk 6]
```diff
@@ -305,8 +323,13 @@ public InputList<Inputs.TopicRuleDynamodbv2Args> Dynamodbv2s
             set => _dynamodbv2s = value;
         }
 
-        [Input("elasticsearch")]
-        public Input<Inputs.TopicRuleElasticsearchArgs>? Elasticsearch { get; set; }
+        [Input("elasticsearches")]
+        private InputList<Inputs.TopicRuleElasticsearchArgs>? _elasticsearches;
+        public InputList<Inputs.TopicRuleElasticsearchArgs> Elasticsearches
```
Comment (thomas11) [inline sdk/dotnet/Iot/TopicRule.cs:328]: This is weird to me as a former user of Elasticsearch. It's the name of the software itself so plural is odd. Something like "ElasticsearchInstances" would be more natural.

**Review by thomas11 — APPROVED** [review]
LGTM except one name. Up to you whether to go through another revision for it.

File: sdk/dotnet/Iot/TopicRule.cs (hunk 6 above)
Comment (iwahbe) [reply sdk/dotnet/Iot/TopicRule.cs:328]: This was fixed by changing `elasticsearches` to `elasticsearch`. This matches other fields in the same object, where it's just the name of a bunch of AWS services. I'll push the commit shortly.

**Review by danielrbradley — APPROVED** [review]

**Pull request merged**